### PR TITLE
Bug 1822349 - tweak commit message replacements

### PR DESCRIPTION
**Pull request by jcristau**
Change the git-filter-repo commit message rewrite step to avoid breaking existing URLs referencing commits, and to remediate some issues caused by rewrites during the previous migrations.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1822349

### CODE REVIEW

File: monorepo-migration/generate-replace-message-expressions.py [hunk 1]
```diff
@@ -57,6 +57,37 @@ def main():
                 )
                 regexes.append(regex)
 
+    # fix up wrong replacements from previous migrations
```
Comment (gbrownmozilla) [inline monorepo-migration/generate-replace-message-expressions.py:60]: This is an impressive list!

**Review by gbrownmozilla — APPROVED** [review]

File: monorepo-migration/generate-replace-message-expressions.py (hunk 1 above)
Comment (JohanLorenzo) [reply monorepo-migration/generate-replace-message-expressions.py:60]: +1 Thank you for finding and fixing these edge cases we missed in the previous migrations!

**Review by JohanLorenzo — APPROVED** [review]
Looks great to me. I look forward to seeing the new history 😃

**Comment by jcristau** [comment]
New rewrite pushed: https://hg.mozilla.org/users/jcristau_mozilla.com/firefox-android-hg/pushloghtml?changeset=8fe4821c3bceb9f7a02a541f9dc751ea9862e873

**Pull request merged**